internal/pkg/sysinfo/probes/probes.go

```diff
@@ -92,11 +97,24 @@ type Probes interface {
 	Probe
 }
 
-// NewProbes returns a new, empty composite probe.
-func NewProbes() Probes {
-	return &probes{}
+// NewRootProbes returns a new, empty composite probe without a path.
+func NewRootProbes() Probes {
+	return &probes{nil, nil}
```

Comment:
if path and probes values are left empty (`return &probes{}`) do they not equal nil by default? (meaning, do we need to set the values in the struct explicitly?)